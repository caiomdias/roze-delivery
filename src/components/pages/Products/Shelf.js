import React, { useEffect, useState } from 'react'
import ProductCard from '../../common/productCard/ProductCard'
import { getProducts } from '../../../services/ZeApi'
import productNotFound from '../../../assets/images/png/product-not-found.png';


const Shelf = (props) => {
	const { categoryId, distributorId } = props
	const [ categoryIdState, setCategoryIdState ] = useState(categoryId || '') 
	const [ distributorIdState, setDistributorId ] = useState(distributorId || '')
	const [ products, setProducts ] = useState([])

	const variables = {
		id: distributorIdState,
		search: '',
		categoryId: categoryIdState
  }
  
  useEffect(() => {
    if(distributorIdState && categoryIdState) {
      getProducts(variables)
        .then(({ data }) => {
          if(data.length != 0) {
            setProducts(data.poc.products)
          }
        })
    }
  }, [])

	return (
    <div className="shelf">
      {
        products.length !== 0
          ? products.map((item, index) => (
            <ProductCard key={index} product={item}/>
          ))
          : <div className="shelf__not-found">
              <img src={productNotFound}></img>
              <p>Putz, n√£o conseguimos encontrar nenhum produto dessa categoria</p>
            </div>
      }
    </div>
	)
};

Shelf.defaultProps = {
	categoryId: '',
	distributorId: '',
}

export default Shelf;
