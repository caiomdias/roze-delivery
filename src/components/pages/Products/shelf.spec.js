import React from 'react';
import renderer from 'react-test-renderer';
import Shelf from './Shelf';

jest.mock('react-router-dom', () => ({
    useLocation: jest.fn().mockReturnValue({
      pathname: '/',
      search: '',
      hash: '',
      state: null,
      key: '5nvxpbdafa',
	}),
	useHistory: jest.fn().mockReturnValue({
		push: jest.fn().mockReturnValue()
	}),
}));

const props = {
	distributorId: '111',
	categoryId: '111'
}

describe("Shelf Component", () => {
	it('Should render correctly Shelf component', () => {  
		const shelfComponent = renderer.create(<Shelf {...props}/>);
		expect(shelfComponent).toMatchSnapshot();
	});
});