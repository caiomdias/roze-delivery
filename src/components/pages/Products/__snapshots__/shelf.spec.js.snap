// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shelf Component Should render correctly Shelf component 1`] = `
<div
  className="shelf"
>
  <div
    className="shelf__not-found"
  >
    <img
      src="test-file-stub"
    />
  </div>
</div>
`;
